### ISSUE TITLE
[rule.unnecessary-stmt] misreport for type assertion

### ISSUE THREAD

**Issue by chen3feng** [mgechev/revive#731]
**Describe the bug**

**To Reproduce**
1. I updated revive `go get -u github.com/mgechev/revive`
2. I run it with the following flags & configuration file:

```go
	infixString := ""
	var content any
	switch val := content.(type) {
	case int32:
		infixString = infixString + string(val)
	}
```

with config:

```toml
[rule.unnecessary-stmt]
```

**Expected behavior**
Should no warning

**Logs**
Report message:

```console
[unnecessary-stmt] switch with only one case can be replaced by an if-then
```

**Desktop (please complete the following information):**
 - OS: [e.g. MacOs 12.3.1]
 - go version go1.14.15 linux/amd64

**Comment by chen3feng** [comment]
It can be rewrite to:

```go
infixString := ""
var content any
val, ok := content.(int32) 
if ok {
	infixString = infixString + string(val)
}
```

sorry for bothering.
